Dockerfile

```diff
@@ -1,28 +1,21 @@
-FROM golang:1.14 AS builder
+FROM golang:1.16 AS builder
```

Comment:
Update the dockerfiles to Go 1.17

Reply:
The Go agent doesn't support 1.17 yet (though we're very close and may even be there before this PR is merged). For now, I'm going to leave it as 1.16. When it comes time to merge, if the agent support 1.17 then I'll update here, otherwise I'll create a new ticket in the 1.17 epic.